src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -457,7 +458,18 @@ func (r *StatefulSetResource) obj() (k8sclient.Object, error) {
 	return ss, nil
 }
 
-func overprovisioned(developerMode bool, limits corev1.ResourceList) []string {
+func prepareAdditionalArguments(
+	developerMode bool, limits, requests corev1.ResourceList,
+) []string {
+	var smp int64 = 1
```

Comment:
Should it run unconstrained if there are no constraints?

Reply:
Agree. Updated.